src/windows.rs

```diff
@@ -12,14 +12,16 @@ extern crate std;
 use crate::Error;
 use core::num::NonZeroU32;
 use std::io;
-use winapi::shared::minwindef::ULONG;
-use winapi::um::ntsecapi::RtlGenRandom;
-use winapi::um::winnt::PVOID;
+
+extern "system" {
+    #[link_name = "SystemFunction036"]
+    fn RtlGenRandom(RandomBuffer: *mut u8, RandomBufferLength: libc::c_ulong) -> u8;
+}
 
 pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
-    // Prevent overflow of ULONG
-    for chunk in dest.chunks_mut(ULONG::max_value() as usize) {
-        let ret = unsafe { RtlGenRandom(chunk.as_mut_ptr() as PVOID, chunk.len() as ULONG) };
+    // Prevent overflow of libc::c_ulong
+    for chunk in dest.chunks_mut(libc::c_ulong::max_value() as usize) {
+        let ret = unsafe { RtlGenRandom(chunk.as_mut_ptr(), chunk.len() as libc::c_ulong) };
```

Comment:
Isn't `c_ulong` on Windows always equals to `u32`? I think we don't need the whole `libc` just for that.

Reply:
That's [correct](https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges). Similarly on macOS/iOS, `SecRandomCopyBytes` is actually [defined to return `i32`](https://developer.apple.com/documentation/security/1399291-secrandomcopybytes), so I also removed that `libc` dependency on macOS/iOS.